lib/pipeline/webpackProcess.js

```diff
@@ -1,208 +1,105 @@
 
 let webpackProcess;
 
 class WebpackProcess extends ProcessWrapper {
   constructor(options) {
     super(options);
-    this.env = options.env;
+    this.webpackConfigName = options.webpackConfigName;
   }
 
-  build(file, importsList, callback) {
-    const self = this;
-    let realCwd;
+  async build(assets, importsList, callback) {
+    try {
+      await this.webpackRun(assets, importsList, callback);
+    } catch (e) {
+      callback(errorMessage(e));
+    }
+  }
 
-    async.waterfall([
-      function changeCwd(next) {
-        realCwd = utils.pwd();
-        process.chdir(fs.embarkPath(''));
-        next();
-      },
+  async webpackRun(assets, importsList, callback) {
+    try {
+      await writeFile(
+        fs.dappPath('.embark/embark-aliases.json'),
+        JSON.stringify(importsList)
+      );
+      await writeFile(
+        fs.dappPath('.embark/embark-assets.json'),
+        JSON.stringify(assets)
+      );
+    } catch (e) {
+      return callback(errorMessage(e));
+    }
 
-      function runWebpack(next) {
-        self.webpackRun(file.filename, {}, true, importsList, true, realCwd, next);
-      },
+    const dappConfigPath = fs.dappPath('webpack.config.js');
+    const defaultConfigPath = fs.embarkPath('lib/pipeline', 'webpack.config.js');
 
-      function changeCwdBack(next) {
-        process.chdir(realCwd);
-        next();
+    let config, configPath;
+    try {
+      if (fs.existsSync(dappConfigPath)) {
+        configPath = dappConfigPath;
+        delete require.cache[configPath];
+      } else {
+        configPath = defaultConfigPath;
       }
-    ], (err) => {
-      process.chdir(realCwd);
-      callback(err);
-    });
-  }
-
-  webpackRun(filename, options, includeModules, importsList, detectErrors, realCwd, callback) {
-    const self = this;
-    glob(fs.dappPath('.embark/versions/*/*'), (err, files) => {
-      let versions;
-      if (err) {
-        console.error(err);
-        versions = [];
+      config = require(configPath);
+      // valid config types: https://webpack.js.org/configuration/configuration-types/
+      // + function that returns a config object
+      // + function that returns a promise for a config object
+      // + array of named config objects
+      // + config object
+      if (typeof config === 'function') {
+        config = await config(this.webpackConfigName);
+      } else if (Array.isArray(config)) {
+        config = config.filter(cfg => cfg.name === this.webpackConfigName);
+        if (!config.length) {
+          return callback(`no webpack config has the name '${this.webpackConfigName}'`);
+        }
+        if (config.length > 1) {
+          console.warn(`detected ${config.length} webpack configs having the name '${this.webpackConfigName}', using the first one`);
+        }
+        config = config[0];
       } else {
-        versions = files;
+        // proceed with the value obtained from require(configPath)
```

Comment:
No need for that comment and `else` IMO.

Reply:
Done.